### ISSUE TITLE
[Question] Custom ImageHandler not running

### ISSUE THREAD

**Issue by FreedCapybara** [xamarin/Xamarin.Forms#1845]
### Description
**tl;dr;**
**My custom ImageHandler isn't running.**

I'm trying to implement a sharing mechanism where the user can pick an image from their picture gallery/library, render it to a view to add some branding overlays, and serialize the view to a bitmap/UIImage for sharing.

Everything works great except that StreamImageSources are still in a state of loading at the time of serialization, resulting in an image of black pixels.

Unfortunately it doesn't look like there's a way to have images load synchronously ([the image handlers call `ConfigureAwait(false)`](https://github.com/xamarin/Xamarin.Forms/blob/cdc405512844671bc3b2c8bd28f583036e5530a2/Xamarin.Forms.Platform.Android/Renderers/StreamImagesourceHandler.cs))

So after poking around the source a little for examples, I've written a ByteArrayImageSource class with a handler that doesn't have any async loading.

_The ImageSource implementation_
```cs
    // Not super sure what all this does, I just kinda ran with whatever StreamImageSource did ¯\_(ツ)_/¯
    // Point is: it has a byte array

    public class ByteArrayImageSource : ImageSource
    {
        public static readonly BindableProperty BytesProperty = BindableProperty.Create("Bytes", typeof(byte[]), typeof(ByteArrayImageSource),
            default(byte[]));

        public virtual byte[] Bytes
        {
            get { return (byte[])GetValue(BytesProperty); }
            set { SetValue(BytesProperty, value); }
        }

        protected override void OnPropertyChanged(string propertyName)
        {
            if (propertyName == BytesProperty.PropertyName)
                OnSourceChanged();
            base.OnPropertyChanged(propertyName);
        }
    }
```

_The handler (iOS)_
```cs
    // Returns a UIImage from a byte array provided by a ByteArrayImageSource

    public class ByteArrayImageSourceHandler : IImageSourceHandler
    {
        public async Task<UIImage> LoadImageAsync(ImageSource imageSource, CancellationToken cancelationToken = default(CancellationToken), float scale = 1f)
        {
            var source = imageSource as ByteArrayImageSource;
            UIImage image = null;

            if (source?.Bytes != null)
            {
                image = await new Task<UIImage>(() => UIImage.LoadFromData(NSData.FromArray(source.Bytes), scale));
            }

            return image;
        }
    }
```

_How I'm trying to glue it together in AssemblyInfo.cs or wherever_
```cs
// This is the order the parameters appear in the XamForms source, though the little method signature thing in VS4Mac says it should be handler -> Source.
// Changing the order of the parameters hasn't made a difference
[assembly: ExportImageSourceHandler(typeof(ByteArrayImageSource), typeof(ByteArrayImageSourceHandler))]
```

Another unfortunate thing is that the project is using a slightly older version of Xamarin.Forms, if that makes a difference (v2.3.4.270 -- can't upgrade because upgrading breaks a lot of stuff and we're not far from releasing).

What am I missing? How do I get my handler to run?

### Expected Behavior
The handler runs and renders a native Bitmap/UIImage to the screen.

### Actual Behavior
Breakpoints in the handler are not hit and the image does not render to the screen.

### Basic Information

- Version with issue: v2.3.4.270
- Last known good version: ¯\_(ツ)_/¯
- IDE: VS4Mac
- Platform Target Frameworks: 
  - iOS:  11.1
  - Android: 8

**Comment by FreedCapybara** [comment]
Thanks for the quick reply.

Clearly I must be missing something because I was able to reproduce the issue with this repo https://github.com/FreedCapybara/XFImageHandlerRepro

**Files of interest:**

_Shared project_
* ImageSources/ByteArrayImageSource.cs
* ViewModels/AboutViewModel.cs

_iOS project_
* Renderers/ByteArrayImageSourceHandler.cs
* Main.cs (includes the assembly definition)

**Steps to reproduce**
1. Launch the app and click `About` on the bottom tabs
2. Click the blue button and select an image from your device (simulators have some images pre-loaded)
3. The image output should display on screen
4. Stop the app, uncomment the line that assigns `RenderOutput` to the ByteArrayImageSource, and comment out the line that assigns `RenderOutput` to ImageSource.FromStream
5. Repeat steps 1 and 2
6. The image does not appear

Thanks for looking into this!

**Comment by samhouts** [comment]
Your handler does run, actually. The problem is that you are awaiting a Task that is never started, so it will never return.

You can just change your handler to this, and it works like a charm:
``` chsarp
image = UIImage.LoadFromData(NSData.FromArray(source.Bytes), scale);
```

Thanks!

**Comment by FreedCapybara** [comment]
Oh man, that was all? I figured I must have had some of the Xamarin details messed up. Thanks for catching my derp!

I create Sainthood as a Service in your honor 🙏 
http://shaas.azurewebsites.net?imageUrl=https://avatars2.githubusercontent.com/u/7827070?s=460&v=4
